### DISCUSSION TITLE
Error in SocialGraph Task and Inactive Social Graph Tab

### DISCUSSION THREAD

**Discussion by dh4340** [cmu-sei/GHOSTS#407] [Q&A]
Hi,

I'm experiencing issues with the `socialgraph` task in GHOSTS SHADOWS. When I run the task, I encounter an error in the logs, and the Social Graph tab never becomes active. Here's a detailed breakdown of the issue:

### Error Details:

In the logs, I see the following entries:

```
2024-09-09 10:32:27.1861|TRACE|ghosts.api.Infrastructure.Services.QueueSyncService|Queue Processed — Machines: 1, Histories: 1, Timeline: 3, Health: 0, Trackables: 0, Webhooks: 0
fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[1]
      An unhandled exception has occurred while executing the request.
      Microsoft.AspNetCore.Server.Kestrel.Core.BadHttpRequestException: Request body too large. The max request body size is 30000000 bytes.
         at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.Http1ContentLengthMessageBody.OnReadStarting()
         ...
```

And further down:

```
fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[1]
      An unhandled exception has occurred while executing the request.
      System.NotImplementedException: The method or operation is not implemented.
         at ghosts.api.Controllers.ViewSocialController.Index() in /app/Ghosts.Api/Controllers/ViewSocialController.cs:line 27
         ...
```

### Symptoms:

1. **Error Message**: The error message indicates that the request body is too large and that a method is not implemented.
2. **Inactive Tab**: The Social Graph tab remains inactive, and no data is displayed.

### Things I Have Tried:

- Verified that the Social Graph service is correctly configured and running.
- Restarted the application and Docker container to ensure no transient issues are affecting the task.

### Question:

Can anyone provide insight into the `Request body too large` error and the `System.NotImplementedException`? How can I address these issues to get the Social Graph tab functioning correctly?

Any help or suggestions would be greatly appreciated!

Thank you!

**Comment by sei-dupdyke** [comment]
First, your issue reports are top notch - A+ — thank you.

The second one is easy:
/app/Ghosts.Api/Controllers/ViewSocialController.cs:line 27

Ahh, no excuses, it really didn't work reliably and so I put [this in there](https://github.com/cmu-sei/GHOSTS/blob/master/src/Ghosts.Api/Controllers/ViewSocialController.cs#L27) until I could get back to it, or someone wanted to fix it up correctly. Sorry about that.

The first one I have not seen. It seems to originate [here in the webhooks](https://github.com/cmu-sei/GHOSTS/blob/master/src/Ghosts.Api/Infrastructure/Services/QueueSyncService.cs#L443) section — can you tell us more about how you are using those and how you have them configured please?

**Reply by dh4340** [reply]
Hi, 

Glad my reports are helpful, I will have to review the setup. But I haven't used webhook at all currently.

**Reply by dh4340** [reply]
I can confirm I am **not** using webhooks and this occurs during a standard setup as per the guide.
